### DISCUSSION TITLE
attempt to index global 'metals_config' (a function value)

### DISCUSSION THREAD

**Discussion by robbyki** [scalameta/nvim-metals#263] [Q&A] [answered]
I'm battling this issue for a while where I get `attempt to index global 'metals_config' (a function value)` when my `init.lua` contains the following snippet and I can't figure out why. Still a little bit new to Lua and going through an exercise of just trying to migrate to nvim-cmp from nvim-compe for completion.
```
metals_config.settings = {
│ showImplicitArguments = true,
│ excludedPackages = { "akka.actor.typed.javadsl", "com.github.swagger.akka.javadsl" },
}
```

**Comment by ivanopagano** [comment, marked as answer]
I think the point is that `metals_config` is considered a function, thus needs be called as `metals_config()`.
This in turn might mean you can't access the `.settings` as a value?

No solution here, sorry, but maybe you can try something starting from this?

**Reply by ivanopagano** [reply]
Probably this is related to a late breaking change documented [here](https://github.com/scalameta/nvim-metals/discussions/253#discussioncomment-1486392)... it might be that's the reason why `metals_config` is seen as a function reference.

Try following the advice there to see if your problem vanishes

**Reply by robbyki** [reply]
thanks. was just testing out the example code as is taken from [here](https://github.com/scalameta/nvim-metals/discussions/39#discussion-82302) for now but a little confused by this error:

**Reply by ckipp01** [reply]
You're exactly correct @ivanopagano .  You can see the context in here: https://github.com/scalameta/nvim-metals/discussions/253

I introduced a breaking change that was sort of necessary. `bare_config` used to just be a table, but it's not a function that returns a table. So you'll just need to add `()`. The error you get is because it thinks your passing a function into the `initialize_or_attatch` function instead of the table it expects.

**Reply by robbyki** [reply]
ahh. ok. sorry. should have noticed the breaking change. thanks alot.

**Reply by ckipp01** [reply]
No worries, I totally thought I had changed it in the minimal config, but realized I didn't. Sorry for that.
